### ISSUE TITLE
Logging from Lucene not collected to Maven logs

### ISSUE THREAD

**Issue by OrangeDog** [dependency-check/DependencyCheck#7120]
**Describe the bug**
These logs from Apache Lucene are going direct to console instead of via Maven's logging.

```
[INFO] Finished Version Filter Analyzer (0 seconds)
Oct 30, 2024 5:07:44 PM org.apache.lucene.store.MemorySegmentIndexInputProvider <init>
INFO: Using MemorySegmentIndexInput with Java 21 or later; to disable start with -Dorg.apache.lucene.store.MMapDirectory.enableMemorySegments=false
Oct 30, 2024 5:07:44 PM org.apache.lucene.internal.vectorization.VectorizationProvider lookup
WARNING: Java vector incubator module is not readable. For optimal vector performance, pass '--add-modules jdk.incubator.vector' to enable Vector API.
[INFO] Created CPE Index (1 seconds)
```

**Version of dependency-check used**
Maven plugin 11.1.0 (on JDK 21)

**Expected behavior**
```
[INFO] Finished Version Filter Analyzer (0 seconds)
[INFO] Using MemorySegmentIndexInput with Java 21 or later; to disable start with -Dorg.apache.lucene.store.MMapDirectory.enableMemorySegments=false
[WARNING] Java vector incubator module is not readable. For optimal vector performance, pass '--add-modules jdk.incubator.vector' to enable Vector API.
[INFO] Created CPE Index (1 seconds)
```

(but also there should not be warnings during standard operation)

**Additional information**
Presumably there is an SLF4J bridge or some internal config missing.

**Comment by OrangeDog** [comment]
N.B. the Apache logging goes to `stderr` while all the other logging goes to `stdout`.
When debug logging is enabled, there's a lot more of it from JCS too.

**Comment by danfred360** [comment]
I can understand why this wouldn't be super high priority but it would be nice to see this resolved since it confuses our developers when there is an actual error that causes the cli to fail. we get reports that there's a java issue instead of what the actual error is 😂

we see these warnings in every run

```
Jun 11, 2025 12:28:49 PM org.apache.lucene.util.HotspotVMOptions <clinit>

WARNING: Lucene cannot optimize algorithms or calculate object sizes for JVMs that are not based on Hotspot or a compatible implementation.

WARNING: A restricted method in java.lang.foreign.Linker has been called

WARNING: java.lang.foreign.Linker::downcallHandle has been called by the unnamed module

WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for this module



Jun 11, 2025 12:28:49 PM org.apache.lucene.store.MemorySegmentIndexInputProvider <init>

INFO: Using MemorySegmentIndexInput and native madvise support with Java 21 or later; to disable start with -Dorg.apache.lucene.store.MMapDirectory.enableMemorySegments=false

Jun 11, 2025 12:28:49 PM org.apache.lucene.internal.vectorization.VectorizationProvider lookup

WARNING: Java runtime is not using Hotspot VM; Java vector incubator API can't be enabled.
```

**Comment by jeremylong** [comment]
I've tried this: https://github.com/dependency-check/DependencyCheck/compare/jul

As this is what I've seen online docs say to route the lucene logs to maven. However, I wasn't able to verify if this actually works.

**Comment by chadlwilson** [comment]
This was fixed via #7979 (still unreleased at time of writing), as part of addressing #7759 . I hadn't seen this issue, or I'd have linked it.

**Comment by chadlwilson** [comment]
Some of the JVM warnings in https://github.com/dependency-check/DependencyCheck/issues/7120#issuecomment-2962679304 are a separate but related issue - as they relate to stuff Lucene does on Java 21+ and cannot be suppressed without either turning off certain Lucene memory index functionality or giving JVM arguments to Maven before it launches its JVM.

This is addressed for the CLI and Docker image at #8003, but within Maven you would probably need you to set `MAVEN_OPTS="-XX:-PrintWarnings"` to get rid of both the linker and jdk vector incubator API warnings (with risk of suppressing other warnings you may be concerned about when running your build).

If only getting the linker warning, `MAVEN_OPTS="--enable-native-access=ALL-UNNAMED"` is probably better.

Or `.mvn/jvm.config` in your project home as noted at https://maven.apache.org/configure.html
